core/client/hdfs/src/main/java/alluxio/hadoop/AbstractFileSystem.java

```diff
@@ -473,15 +473,23 @@ public synchronized void initialize(URI uri, org.apache.hadoop.conf.Configuratio
     mStatistics = statistics;
     mUri = URI.create(mAlluxioHeader);
 
+    // take the URI properties, hadoop configuration, and given Alluxio configuration and merge
+    // all three into a single object.
     Map<String, Object> uriConfProperties = getConfigurationFromUri(uri);
-
+    Map<String, Object> hadoopConfProperties =
+        HadoopConfigurationUtils.getConfigurationFromHadoop(conf);
+    LOG.info("Creating Alluxio configuration from Hadoop configuration {}, uri configuration {}",
+        hadoopConfProperties, uriConfProperties);
     AlluxioProperties alluxioProps =
         (alluxioConfiguration != null) ? alluxioConfiguration.copyProperties()
             : ConfigurationUtils.defaults();
-    LOG.info("Creating Alluxio configuration from Hadoop configuration {}, uri configuration {}",
-        conf, uriConfProperties);
-    AlluxioConfiguration alluxioConf = mergeConfigurations(uriConfProperties, conf, alluxioProps);
-    mAlluxioConf = alluxioConf;
+    // Merge relevant Hadoop configuration into Alluxio's configuration.
+    alluxioProps.merge(hadoopConfProperties, Source.RUNTIME);
```

Comment:
let's put all configuration merge in the same place, rather than distributed in multiple methods